### PR TITLE
Upgraded to version 3 of companion

### PR DESCRIPTION
**Pull request by Bambii556**
These changes are for upgrading the module to the latest version of the companion.
The Actions has been updated to now query the services of the stream that are active, and starting the stream had to be split between Main and Sub streams
Also added User and Password fields to the Config for API Auth

### CODE REVIEW

File: .vscode/settings.json
Comment (dnmeid) [inline .vscode/settings.json:1]: Please don't include in repo

File: companion/manifest.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	"name": "kiloview-encoder",
+	"shortname": "kiloview",
+	"description": "Module to control Kiloview Encoders",
+	"version": "1.0.1",
+	"license": "MIT",
+	"repository": "git+https://github.com/bitfocus/companion-module-kiloview-encoder.git",
+	"bugs": "https://github.com/bitfocus/companion-module-kiloview-encoder/issues",
+	"maintainers": [
+		{
+			"name": "",
```
Comment (dnmeid) [inline companion/manifest.json]: Do you want to maintain this module?

File: package.json [hunk 2]
```diff
@@ -1,7 +1,8 @@
 {
   "name": "kiloview-encoder",
-  "version": "1.0.0",
+  "version": "1.0.1",
```
Comment (dnmeid) [inline package.json]: Please bump to 2.0.0
https://github.com/bitfocus/companion/wiki/Versioning

File: package.json [hunk 3]
```diff
@@ -1,7 +1,8 @@
...
   "api_version": "1.0.0",
+	"type": "module",
   "keywords": [
```
Comment (dnmeid) [inline package.json]: Please remove unused properties, which are now in manifest.

File: package.json [hunk 4]
```diff
@@ -11,21 +12,22 @@
...
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "format": "prettier -w .",
+    "lint:raw": "eslint --ext .ts --ext .js --ignore-pattern dist --ignore-pattern pkg",
+    "lint": "yarn lint:raw ."
   },
   "author": "ifmx-dev",
   "license": "MIT",
   "homepage": "https://github.com/bitfocus/companion-module-kiloview-encoder#readme",
```
Comment (dnmeid) [inline package.json]: -> manifest

File: src/presets.js
Comment (dnmeid) [inline src/presets.js:1]: preset object is not fully converted to new format. E.g. steps, Down/Up-Actions, actionId...
https://github.com/bitfocus/companion-module-base/wiki/Upgrading-a-module-built-for-Companion-2.x#7-updating-your-presets

**Review by dnmeid — CHANGES_REQUESTED** [review]
Thanks for the update!
Some minor changes needed.

File: .vscode/settings.json
Comment (Bambii556) [reply .vscode/settings.json:1]: ok removed and added to gitignore

File: package.json (hunk 2 above)
Comment (Bambii556) [reply package.json]: ok version bumped and thanks for the docs, seems i missed that part

File: companion/manifest.json (hunk 1 above)
Comment (Bambii556) [reply companion/manifest.json]: i actually copied that file as a template, but not sure who the maintainer was?

File: package.json (hunk 3 above)
Comment (Bambii556) [reply package.json]: Removed all duplicate entries in package.json, except type is used by the project, so it has to be there

File: src/presets.js
Comment (Bambii556) [reply src/presets.js:1]: ok updated the presets, but had to remove one, as the actions have options that only gets populated on connection.

Comment (dnmeid) [reply src/presets.js:1]: That is not a problem at all. You can generate presets dynamically the same way as dropdown choices. When the device connects and you know the needed data, you can generate a new set of actions and presets and push it to the core.
Only thing to take care of, is that when a user applies the preset, this option data will be stored in the database and if he later connects to a different device and the option is not valid any more this should not crash the module. For example if a device has some channels and those channels have names, it is better to use the channel number as the index than the channel name. If you use the channel name and the name changes the option will be invalid. If you use the number and the name changes the number will stay the same.

**Review by dnmeid — APPROVED** [review]
Left a comment about the presets. Also I see that the recording start/stop is solved with steps. As the recording state is being fetched anyway for feedback, it would be much better to introduce a `toggle` choice to the recording action. Then the steps can be replaced by a single toggle recording action and there are no more problems when you have the same button on multiple pages or the recording state changes from a reason outside of companion.

Tell me if you want to do that changes, then I wait with the merge.

File: src/actions.js [hunk 5]
```diff
@@ -29,6 +29,13 @@ export function getActionDefinitions(self) {
 				let method = 'startRecord'
 				if (action.options.status === 'stop') {
 					method = 'stopRecord'
+				} else if (action.options.status === 'toggle') {
+					const stream = self.cache.streams[feedback.options.stream]
```
Comment (dnmeid) [inline src/actions.js]:
```suggestion
					const stream = self.cache.streams[action.options.stream]
```

**Review by dnmeid — CHANGES_REQUESTED** [review]

**Pull request merged**